### ISSUE TITLE
JSON variables cannot contain arrays

### ISSUE THREAD

**Issue by kmcclellan** [ChilliCream/graphql-platform#6182]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

The `JSON` scalar type should let you pass arbitrary JSON to an endpoint via the GraphQL variables. This only works if the JSON doesn't contain any arrays.

OK:
```json
{
  "input": {
    "hello": "world"
  }
}
```

Fails:
```json
{
  "input": {
    "hello": [ "earth", "mars" ]
  }
}
```

### Steps to reproduce

1. Run the web app:

```csharp
using HelloGql;
using HotChocolate.Execution;
using HotChocolate.Execution.Instrumentation;
using System.Text.Json;

var builder = WebApplication.CreateBuilder(args);

builder.Services.AddGraphQLServer()
    .AddDiagnosticEventListener<RequestErrorLogger>()
    .AddQueryType<QueryController>();

var app = builder.Build();

app.MapGraphQL();
app.Run();

namespace HelloGql
{
    public class QueryController
    {
        public JsonElement EchoJson(JsonElement json)
        {
            return json;
        }
    }

    internal class RequestErrorLogger : ExecutionDiagnosticEventListener
    {
        private readonly ILogger<RequestErrorLogger> _logger;

        public RequestErrorLogger(ILogger<RequestErrorLogger> logger)
        {
            _logger = logger;
        }

        // These errors don't get logged by the library for some reason.
        public override void RequestError(IRequestContext context, Exception exception)
        {
            if (exception is GraphQLException gqlEx)
            {
                // Aggregate exception will print inner exceptions.
                exception = new AggregateException(
                    gqlEx.Errors.Select(x => x.Exception).OfType<Exception>().Prepend(exception));
            }

            _logger.LogError(exception, "Exception while executing GraphQL request");
        }
    }
}
```

2. Enter variables in Banana Cake Pop:

```json
{
  "input": {
    "hello": [ "earth", "mars" ]
  }
}
```

3. Execute query in Banana Cake Pop:

```
query ($input: JSON!) {
  echoJson(json: $input)
}
```

### Relevant log output

```shell
{
  "errors": [
    {
      "message": "Variable `input` got an invalid value.",
      "locations": [
        {
          "line": 1,
          "column": 8
        }
      ],
      "extensions": {
        "code": "HC0016",
        "variable": "input"
      }
    }
  ]
}

fail: HelloGql.RequestErrorLogger[0]
      Exception while executing GraphQL request
      System.AggregateException: One or more errors occurred. (Variable `input` got an invalid value.) ('}' is invalid without a matching open.)
       ---> HotChocolate.GraphQLException: Variable `input` got an invalid value.
         at HotChocolate.Execution.Processing.VariableCoercionHelper.CoerceVariableValue(VariableDefinitionNode variableDefinition, IInputType variableType, Object value)
         at HotChocolate.Execution.Processing.VariableCoercionHelper.CoerceVariableValues(ISchema schema, IReadOnlyList`1 variableDefinitions, IReadOnlyDictionary`2 values, IDictionary`2 coercedValues)
         at HotChocolate.Execution.Pipeline.PipelineTools.CoerceVariables(IRequestContext context, VariableCoercionHelper coercionHelper, IReadOnlyList`1 variableDefinitions)
         at HotChocolate.Execution.Pipeline.OperationVariableCoercionMiddleware.InvokeAsync(IRequestContext context)
         at HotChocolate.Execution.Pipeline.OperationResolverMiddleware.InvokeAsync(IRequestContext context)
         at HotChocolate.Execution.Pipeline.OperationComplexityMiddleware.InvokeAsync(IRequestContext context)
         at HotChocolate.Execution.Pipeline.OperationCacheMiddleware.InvokeAsync(IRequestContext context)
         at HotChocolate.Execution.Pipeline.DocumentValidationMiddleware.InvokeAsync(IRequestContext context)
         at HotChocolate.Execution.Pipeline.DocumentParserMiddleware.InvokeAsync(IRequestContext context)
         at HotChocolate.Execution.Pipeline.DocumentCacheMiddleware.InvokeAsync(IRequestContext context)
         at HotChocolate.Execution.Pipeline.TimeoutMiddleware.InvokeAsync(IRequestContext context)
         at HotChocolate.Execution.Pipeline.ExceptionMiddleware.InvokeAsync(IRequestContext context)
         --- End of inner exception stack trace ---
       ---> (Inner Exception #1) System.InvalidOperationException: '}' is invalid without a matching open.
         at System.Text.Json.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource, Int32 currentDepth, Int32 maxDepth, Byte token, JsonTokenType tokenType)
         at System.Text.Json.Utf8JsonWriter.ValidateEnd(Byte token)
         at System.Text.Json.Utf8JsonWriter.WriteEndSlow(Byte token)
         at System.Text.Json.Utf8JsonWriter.WriteEnd(Byte token)
         at System.Text.Json.Utf8JsonWriter.WriteEndObject()
         at HotChocolate.Types.JsonType.JsonFormatter.JsonFormatterVisitor.Leave(ListValueNode node, JsonFormatterContext context)
         at HotChocolate.Language.Visitors.SyntaxVisitor`1.Visit[TNode,TParent](TNode node, TParent parent, TContext context)
         at HotChocolate.Language.Visitors.SyntaxVisitor`1.VisitChildren(ObjectFieldNode node, TContext context)
         at HotChocolate.Language.Visitors.SyntaxVisitor`1.Visit[TNode,TParent](TNode node, TParent parent, TContext context)
         at HotChocolate.Language.Visitors.SyntaxVisitor`1.VisitChildren(ObjectValueNode node, TContext context)
         at HotChocolate.Language.Visitors.SyntaxVisitor`1.Visit[TNode,TParent](TNode node, TParent parent, TContext context)
         at HotChocolate.Language.Visitors.SyntaxVisitor`1.Visit(ISyntaxNode node, TContext context)
         at HotChocolate.Types.JsonType.JsonFormatter.Format(IValueNode node)
         at HotChocolate.Types.JsonType.ParseLiteral(IValueNode valueSyntax)
         at HotChocolate.Types.InputParser.ParseLeaf(IValueNode resultValue, ILeafType type, Path path, IInputFieldInfo field)
         at HotChocolate.Types.InputParser.ParseLiteralInternal(IValueNode value, IType type, Path path, Int32 stack, Boolean defaults, IInputFieldInfo field)
         at HotChocolate.Types.InputParser.ParseLiteralInternal(IValueNode value, IType type, Path path, Int32 stack, Boolean defaults, IInputFieldInfo field)
         at HotChocolate.Types.InputParser.ParseLiteral(IValueNode value, IType type, Path path)
         at HotChocolate.Execution.Processing.VariableCoercionHelper.CoerceVariableValue(VariableDefinitionNode variableDefinition, IInputType variableType, Object value)<---
```

### Version

13.0.5

**Comment by michaelstaib** [comment]
Can you retest with 13.1

**Comment by kmcclellan** [comment]
> Can you retest with 13.1

Yes fixed in 13.1 🕺 Thanks!
